src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -128,18 +155,49 @@ public static TripInstruction alignTravelerToTrip(
         return null;
     }
 
+    /**
+     * Align the traveler's position to the nearest transit stop or destination.
+     */
+    @Nullable
+    public static TripInstruction alignTravelerToTransitTrip(TravelerPosition travelerPosition) {
+        Locale locale = travelerPosition.locale;
+        Leg expectedLeg = travelerPosition.expectedLeg;
+        String finalStop = expectedLeg.to.name;
+
+        if (isApproachingEndOfLeg(travelerPosition)) {
+            return new GetOffHereTransitInstruction(finalStop, locale);
+        }
+
+        Place nextStop = snapToWaypoint(travelerPosition, getIntermediateAndLastStop(expectedLeg), true);
+        if (nextStop != null) {
+            int stopsRemaining = stopsUntilEndOfLeg(nextStop, expectedLeg);
+            double distance = getDistance(travelerPosition.currentPosition, new Coordinates(nextStop));
+            if (stopsRemaining == 1 && distance <= TRIP_INSTRUCTION_UPCOMING_RADIUS && !isPositionPastStep(travelerPosition, nextStop) || stopsRemaining == 0) {
+                return new GetOffNextStopTransitInstruction(finalStop, locale);
```

Comment:
Should we add a distance check to this? What if it's an express but and the next stop is after a long run on a freeway or something like that?

Reply:
Oh nevermind I see this has been thought of above